### PR TITLE
ARROW-3319: [GLib] Add align() to GArrowInputStream and GArrowOutputStream

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by kou** [comment]
@shiro615 Can you review this?

**Comment by wesm** [comment]
It occurred to me that we could make these methods `OutputStream::Align` or `InputStream::Align` instead of top level functions. Do you have an opinion?

**Comment by kou** [comment]
It's natural that `InputStream::Align()` and `OutputStream::Align()` in object oriented API.
But the current `AlignStream()` in `ipc/` isn't bad because this functionality is for IPC.

So both `InputStream::Align()` and `AlignStream()` in `ipc/` are natural to me.

In Arrow GLib API, I prefer object oriented API because it's natural in Ruby. (Arrow GLib API is mapped to Arrow Ruby API automatically.)

File: c_glib/test/test-buffer-input-stream.rb [hunk 1]
```diff
@@ -22,4 +22,20 @@ def test_read
...
+    buffer_input_stream = Arrow::BufferInputStream.new(buffer)
+    buffer_input_stream.advance(6)
+    read_buffer = buffer_input_stream.read(5)
+    assert_equal("World", read_buffer.data.to_s)
+  end
+
+  def test_align
+    buffer = Arrow::Buffer.new("Hello World")
+    buffer_input_stream = Arrow::BufferInputStream.new(buffer)
+    buffer_input_stream.advance(8)
```
Comment (shiro615) [inline c_glib/test/test-buffer-input-stream.rb]: Perhaps should we test with `align ()` ?

Comment (kou) [reply c_glib/test/test-buffer-input-stream.rb]: Oh! Good catch!
I've fixed this.

**Review by shiro615 — APPROVED** [review]
LGTM.

**Comment by kou** [comment]
Thanks! I'll merge this.

**Pull request closed without merging**